dispatcher.go

```diff
@@ -85,52 +105,80 @@ func (b *backendNF) readFromServer() {
+				//update GnbId for each gnb connection
+				ran, _ := context.Sctplb_Self().RanFindByGnbIp(response.GnbIpAddr)
+				if ran != nil && response.GnbId != "" {
+					ran.SetRanId(response.GnbId)
+				}
+			} else if response.Msgtype == gClient.MsgType_REDIRECT_MSG {
+				//TODO
 			} else {
-				b.sctp.Write(response.Msg)
+				//fetch ran connection based on GnbId
+				if response.GnbId == "" {
+					log.Printf("Received null GnbId from backend NF")
+				} else {
+					ran, _ := context.Sctplb_Self().RanFindByGnbIp(response.GnbIpAddr)
+					if ran != nil {
+						ran.SetRanId(response.GnbId)
+						ran.Conn.Write(response.Msg)
+					} else {
+						log.Printf("Couldn't fetch sctp connection with GnbId: %v", response.GnbId)
+					}
+				}
 			}
 		}
 	}
 }
 
 func (b *backendNF) connectToServer() {
-	target := fmt.Sprintf("%s:%d",b.address, SimappConfig.Configuration.SctpGrpcPort)
+	target := fmt.Sprintf("%s:%d", b.address, SimappConfig.Configuration.SctpGrpcPort)
 
 	fmt.Println("Connecting to target ", target)
 
 	var err error
 	b.conn, err = grpc.Dial(target, grpc.WithInsecure())
 
 	if err != nil {
-		log.Fatalf("did not connect: %s", err)
+		fmt.Println("did not connect: ", err)
+		return
 	}
 
 	//b.conn = conn
 	b.gc = gClient.NewNgapServiceClient(b.conn)
 
-	stream, err := b.gc.HandleMessage(context.Background())
+	stream, err := b.gc.HandleMessage(ctxt.Background())
 	if err != nil {
-		log.Fatalf("openn stream error %v", err)
+		log.Println("openn stream error ", err)
+		return
 	}
 
 	b.stream = stream
+	b.state = true
 	for {
-		req := gClient.Message{}
-		req.VerboseMsg = "Hello From SCTP LB !"
-		req.Msgtype = gClient.MsgType_INIT_MSG
-		req.SctplbId = os.Getenv("HOSTNAME")
-
-		if err := stream.Send(&req); err != nil {
-			log.Fatalf("can not send %v", err)
-		}
-		log.Printf("Send Request message : %v", req)
-		response, err := stream.Recv()
-		if err != nil {
-			log.Println("Response from server: error ", err)
-			b.state = false
-		} else {
-			log.Printf("Init Response from Server %s server: %s", response.AmfId, response.VerboseMsg)
-			b.state = true
-		}
+		context.Sctplb_Self().RanPool.Range(func(key, value interface{}) bool {
+			req := gClient.SctplbMessage{}
+			req.VerboseMsg = "Hello From SCTP LB !"
+			req.Msgtype = gClient.MsgType_INIT_MSG
+			req.SctplbId = os.Getenv("HOSTNAME")
+			candidate := value.(*context.Ran)
+			req.GnbIpAddr = candidate.GnbIp
+			if candidate.RanId != nil {
+				req.GnbId = *candidate.RanId
+			}
+			if err := stream.Send(&req); err != nil {
+				log.Println("can not send: ", err)
+			}
+			log.Printf("Send Request message : %v", req)
+			response, err := stream.Recv()
+			if err != nil {
+				log.Println("Response from server: error ", err)
+				b.state = false
+			} else {
+				log.Printf("Init Response from Server %s server: %s", response.AmfId, response.VerboseMsg)
+				b.state = true
+			}
+			return false
```

Comment:
Do we want to send hello from sctplb for every RAN that has connected. ?

Reply:
yes, addressed your comments in latest code